table-api/src/main/java/io/deephaven/api/agg/util/AggCountType.java

```diff
@@ -0,0 +1,319 @@
+        }
+    }
+
+    public static LongCountFunction getLongCountFunction(final AggCountType countType) {
+        switch (countType) {
+            case ALL:
+                return value -> true;
+            case NON_NULL:
+            case FINITE:
+                return value -> value != QueryConstants.NULL_LONG;
+            case NULL:
+                return value -> value == QueryConstants.NULL_LONG;
+            case POSITIVE:
+                return value -> value > 0; // NULL_LONG is negative
+            case ZERO:
+                return value -> value == 0;
+            case NEGATIVE:
+                return value -> value != QueryConstants.NULL_LONG && value < 0;
+            case NAN:
+            case INFINITE:
+                return value -> false;
+            default:
+                throw new UnsupportedOperationException(
+                        "Unsupported count type: " + countType + " for long data type");
+        }
+    }
+
+    public static FloatCountFunction getFloatCountFunction(final AggCountType countType) {
+        switch (countType) {
+            case ALL:
+                return value -> true;
+            case NON_NULL:
+                return value -> value != QueryConstants.NULL_FLOAT;
+            case NULL:
+                return value -> value == QueryConstants.NULL_FLOAT;
+            case POSITIVE:
+                return value -> value > 0; // NULL_FLOAT is negative
+            case ZERO:
+                return value -> value == 0;
+            case NEGATIVE:
+                return value -> value != QueryConstants.NULL_FLOAT && value < 0;
+            case NAN:
+                return Float::isNaN;
+            case INFINITE:
+                return Float::isInfinite;
+            case FINITE:
+                return value -> value != QueryConstants.NULL_FLOAT && Float.isFinite(value);
+            default:
+                throw new UnsupportedOperationException(
+                        "Unsupported count type: " + countType + " for float data type");
+        }
+    }
+
+    public static DoubleCountFunction getDoubleCountFunction(final AggCountType countType) {
+        switch (countType) {
+            case ALL:
+                return value -> true;
+            case NON_NULL:
+                return value -> value != QueryConstants.NULL_DOUBLE;
+            case NULL:
+                return value -> value == QueryConstants.NULL_DOUBLE;
+            case POSITIVE:
+                return value -> value > 0; // NULL_DOUBLE is negative
+            case ZERO:
+                return value -> value == 0;
+            case NEGATIVE:
+                return value -> value != QueryConstants.NULL_DOUBLE && value < 0;
+            case NAN:
+                return Double::isNaN;
+            case INFINITE:
+                return Double::isInfinite;
+            case FINITE:
+                return value -> value != QueryConstants.NULL_DOUBLE && Double.isFinite(value);
```

Comment:
I think a common case for this would be non-null non-nan; but maybe Finite is actually what people want?  We probably need to have a definition where the constant is defined in the Javadoc and mention the behavior of nans, etc.

Also non-zero.